packages/cozy-client/src/CozyClient.spec.js

```diff
@@ -151,6 +151,22 @@ describe('CozyClient', () => {
         ])
       ])
     })
+
+    it('should handle empty associations', () => {
+      const NEW_TODO = {
+        _type: 'io.cozy.todos',
+        label: 'Buy RAM',
+        attachments: [{ _id: 12345, _type: 'io.cozy.files' }]
+      }
+      const mutation = client.getDocumentSavePlan(NEW_TODO, {
+        attachments: []
+      })
+      expect(Array.isArray(mutation)).toBe(false)
+      expect(mutation).toEqual({
+        mutationType: 'CREATE_DOCUMENT',
+        document: NEW_TODO
```

Comment:
I do not understand, here it means that attachments would not be removed from the TODO even if we tell `getDocumentSavePlan` to do it ?

Reply:
No, AFAIK, `getDocumentSavePlan` is only used to add new relationships / attachments, not to remove them. I'm not sure how that would be done at the moment, but it doesn't seem to go through `getDocumentSavePlan`.